core/src/main/java/de/danoeh/antennapod/core/feed/LocalFeedUpdater.java

```diff
@@ -0,0 +1,109 @@
+package de.danoeh.antennapod.core.feed;
+
+import android.content.Context;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+import android.webkit.MimeTypeMap;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import de.danoeh.antennapod.core.storage.DBTasks;
+
+public class LocalFeedUpdater {
+
+    private static long getFileDuration(File f) {
+        MediaMetadataRetriever mediaMetadataRetriever = new MediaMetadataRetriever();
+        mediaMetadataRetriever.setDataSource(f.getAbsolutePath());
+        String durationStr = mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION);
+        return Long.parseLong(durationStr);
+    }
+
+    /** Starts the import process. */
+    public static void startImport(Uri uri, Context context) {
+        File f = new File(uri.getPath());
+        if (!f.isDirectory()) {
+            throw new RuntimeException("invalid path");
+        } else {
+            startImportDirectory(uri, context);
+        }
+    }
+
+    private static void startImportDirectory(Uri uri, Context context) {
```

Comment:
Please split this into two methods. One (that gets Uri and context) that just creates the `Feed` and one that takes a feed and then adds the files. Then you can use the one that takes the feed `DBTasks`. It might also be a good idea to add a method `isLocalFeed` to the `Feed` class. This can be used when updating and later when deciding what to do when deleting the feed.